src/include/miopen/generic_search.hpp

```diff
@@ -357,6 +367,20 @@ auto GenericSearch(const Solver s, const Context& context, const AnyInvokeParams
         {
             current_solution = s.GetSolution(context, current_config, true);
 
+            if(compile_and_run == "0")
+            {
+                std::vector<KernelInfo> kernels;
+                for(auto&& kernel : current_solution.construction_params)
+                {
+                    if(profile_h.HasProgram(kernel.kernel_file, kernel.comp_options))
+                        continue;
+                    kernels.push_back(kernel);
+                }
+
+                std::vector<Program> programs = PrecompileKernels(profile_h, kernels);
+                continue;
```

Comment:
[Quality] You may wish adding a new monitoring method to Heartbeat and using it here, to show signs of life when log level < 6.